infra/lib/ci-cluster.ts

```diff
@@ -34,38 +37,59 @@ export class CICluster extends Construct {
 
     const clusterVersion = eks.KubernetesVersion.V1_23;
 
-    const mngProps: blueprints.MngClusterProviderProps = {
-      minSize: 2,
-      maxSize: 8,
-      desiredSize: 2,
-      diskSize: 150,
-      version: clusterVersion,
-      instanceTypes: [
-        ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.XLARGE8),
+    const subnetTagPattern = `${STACK_NAME}/${CLUSTER_CONSTRUCT_NAME}/${CLUSTER_NAME}/${CLUSTER_NAME}-vpc/PrivateSubnet*`;
+    const securityGroupTagPattern = "kubernetes.io/cluster/" + CLUSTER_NAME;
+
+    const karpenterAddonProps = {
+      requirements: [
+          {
+            key: 'node.kubernetes.io/instance-type',
+            op: "In" as const,
+            vals: ['m5.xlarge','m5.2xlarge','m5.4xlarge','m5.8xlarge'],
+          },
+          {
+            key: 'kubernetes.io/arch',
+            op: "In" as const,
+            vals: ['amd64']
+          },
+          {
+            key: 'karpenter.sh/capacity-type',
+            op: "In" as const,
+            vals: ['on-demand']
+          },
       ],
-      amiType: eks.NodegroupAmiType.AL2_X86_64,
-      nodeGroupCapacityType: eks.CapacityType.ON_DEMAND,
-    };
+      subnetTags: { "Name": subnetTagPattern },
+      securityGroupTags: { securityGroupTagPattern : "owned" },
+      taints: [{
+        key: "workload",
+        value: "test",
+        effect: "NoSchedule" as const,
+      }],
+      amiFamily: "AL2" as const,
+      consolidation: { enabled: true },
+      ttlSecondsUntilExpired: 2592000,
+      weight: 20,
+      interruptionHandling: true,
+    }
```

Comment:
Could we also add total limits so that the cluster doesn't exceed too many resources. See https://karpenter.sh/preview/concepts/provisioners/#speclimitsresources

I'm not sure what an appropriate value would be, but this could be updated later if we are restricted?

Reply:
I do not see any support for spec.limits in the cdk-eks-blueprints code for the KarpenterAddOn:  https://github.com/aws-quickstart/cdk-eks-blueprints/blob/01b2bd9f488ebc2bd0e159c122591437745af178/lib/addons/karpenter/index.ts#L262
So I suspect we're going to have to just add those tweaks post-hoc by `kubectl edit...`.